### PR TITLE
Proposed Sales Location page

### PR DESCRIPTION
**Pull request by alicia-mbta**
#### Summary of changes
**Asana Tickets:**
[Upload new data source](https://app.asana.com/0/555089885850811/1199957984734799)
[Update the location cards](https://app.asana.com/0/555089885850811/1199957984734798)
[Duplicate the existing tool into a new, unique instance](https://app.asana.com/0/555089885850811/1199957984734796)
[Restyle the new tool](https://app.asana.com/0/555089885850811/1199957984734797)

Gets data from ArcGIS depending on location.

### CODE REVIEW

**Comment by hannahpurcell** [comment]
Looks so great!  At first glance, I noticed one need-to-fix and a couple things that are cosmetic and not a priority for Monday.  I'll note them both, but maybe we could make another "polish" ticket.

Important:  the search button on the input box is a broken link.  (If I click my address from the dropdown, no problem.  But if I click the magnifying glass again later, the url changes and goes to a dead page.)

To do later:
- The address probably shouldn't convert to lat/lon coords in the input box.  It should stay in the format that the user input.  (Maybe that's coordinates, but probably it's an address.)
- The grid of locations can be styled as a "masonry grid"

File: apps/site/lib/site_web/templates/fare_transformation/_nearby_locations.html.eex
Comment (thecristen) [inline apps/site/lib/site_web/templates/fare_transformation/_nearby_locations.html.eex:6]: If you remove `display: flex` from the parent then this wrapper doesn't seem to be needed.

File: apps/site/lib/site_web/templates/fare_transformation/_nearby_locations.html.eex
Comment (thecristen) [inline apps/site/lib/site_web/templates/fare_transformation/_nearby_locations.html.eex:8]: eh, let's make another `c-location-card` class for this instead of appropriating this one.

File: apps/site/lib/site_web/templates/fare_transformation/_nearby_locations.html.eex
Comment (thecristen) [inline apps/site/lib/site_web/templates/fare_transformation/_nearby_locations.html.eex:19]: Nice

**Review by thecristen — APPROVED** [review]
Looks good! I had a few comments but I don't know if any should be dealbreakers. I'd get approval from design and/or product before merging.

**Comment by thecristen** [comment]
+1 to Hannah's observations, also the "use my current location" link doesn't work as expected.

**Comment by hannahpurcell** [comment]
Another medium-priority task for the "Do Later" ticket:  locations should be sorted by distance

File: apps/site/lib/site_web/controllers/fare_transformation_controller.ex
Comment (hannahpurcell) [inline apps/site/lib/site_web/controllers/fare_transformation_controller.ex:27]: ⚠️  The `Nearby.nearby_with_varying_radius_by_mode(position)` isn't catching all possible PSLs.

Search "Boston Landing", and the PSL page will only return one option on Arthur St @ Guest St, but there is a PSL pin in the ArcGIS data directly at Boston Landing.  When I hardcode the logic in `ProposedLocationsby_stop_id()` to also include the id "Boston Landing", the PSL shows up on the page.  So there's something about the `Nearby` logic that isn't finding this station that is very close by

Comment (hannahpurcell) [reply apps/site/lib/site_web/controllers/fare_transformation_controller.ex:27]: Does the query to ArcGIS need to be based on stops, or can it be based on lat-lon?  

I believe ArcGIS can be queried to "get all PSLs within 10 miles of this location."  If we use this query format and just pass in the user input as the central location, than we can skip this `Nearby` step

File: apps/fares/lib/proposed_locations.ex
Comment (thecristen) [inline apps/fares/lib/proposed_locations.ex:14]: Why not do an initial getting and parsing of the entire dataset, and then later just filter those results by the distance? Why involve stop_ids?

File: apps/fares/lib/proposed_locations/location.ex
Comment (thecristen) [inline apps/fares/lib/proposed_locations/location.ex:3]: We might want to limit this just to fields we care about, for simplicity - we probably aren't going to use FFFVMUpdat, LFFVMUpdat, RSTUpdat, why_sn.

File: apps/fares/lib/proposed_locations.ex
Comment (alicia-mbta) [reply apps/fares/lib/proposed_locations.ex]: I just thought getting _all_ the data was a lot, that's why my idea was to grab nearby stops and query the ArcGIS data with that. However, @hannahpurcell thought of an even better query which returns nearby locations by lat/lon! 🎉 so that's what I did instead.

**Comment by alicia-mbta** [comment]
> +1 to Hannah's observations, also the "use my current location" link doesn't work as expected.

I had to really make sure location services was enabled in the browser; also it seems like the changes don't really occur unless you restart it 😕

File: apps/fares/lib/proposed_locations.ex
Comment (thecristen) [inline apps/fares/lib/proposed_locations.ex:11]: This might be too small a limit for the more suburban areas.... would it make sense to do a very large limit and limit the results shown to the (nine? I don't know) closest ones?

Comment (alicia-mbta) [reply apps/fares/lib/proposed_locations.ex:11]: Totally -- I'll change that.

**Comment by hannahpurcell** [comment]
Clicking the 🔍 doesn't go to a dead link anymore, but it also never returns any results (even if there are results)

**Review by hannahpurcell — APPROVED** [review]

**Pull request merged**
